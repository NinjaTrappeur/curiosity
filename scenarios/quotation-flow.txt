reset
user create alice a alice@example.com --accept-tos

user create susie s susie@example.com --accept-tos
user update USER-2 Susie "I'm Susie, the Seller in the quotation-flow scenario."

user create charlie c charlie@example.com --accept-tos
user update USER-3 Charlie "I'm Charlie, a Client in the quotation-flow scenario."

as susie
forms quotation new
# forms quotation edit
forms quotation submit TBPJLIUG

as charlie
quotation sign QUOT-1

as susie
invoice emit --from ORD-1

as system
reminder send INV-1
payment match INV-1
