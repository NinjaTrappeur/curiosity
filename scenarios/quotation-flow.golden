1: reset
State is now empty.
2: user signup alice a alice@example.com --accept-tos
User created: USER-1
Signup confirmation email enqueued: EMAIL-1
4: user signup mila s mila@example.com --accept-tos
User created: USER-2
Signup confirmation email enqueued: EMAIL-2
5: user update USER-2 Mila "I'm Mila, the Seller in the quotation-flow scenario."
User updated: USER-2
7: user signup charlie c charlie@example.com --accept-tos
User created: USER-3
Signup confirmation email enqueued: EMAIL-3
8: user update USER-3 Charlie "I'm Charlie, a Client in the quotation-flow scenario."
User updated: USER-3
10: entity create one "One S.A." 100200300 BE0100200300
Legal entity created: LENT-1
11: entity set-supervised one
Legal entity updated: one
12: entity set-host one
Legal entity updated: one
14: unit create alpha Alpha
Business entity created: BENT-1
15: unit update alpha "The first business unit of the prototype, Alpha is run by @mila."
Business unit updated: alpha
16: unit link-user alpha USER-2 --holder
Business unit updated: alpha
18: as mila
Modifying default user.
19: forms quotation new --client charlie --seller-entity one --seller-unit alpha --buyer-entity one --buyer-unit alpha
Quotation form created: TBPJLIUG
20: forms quotation validate TBPJLIUG
Quotation form is valid.
22: forms quotation submit TBPJLIUG
Quotation form validated.
Quotation created: QUOT-1
Quotation sent to client: QUOT-1
24: as charlie
Modifying default user.
25: quotation sign QUOT-1
Order created: ORD-1
27: as mila
Modifying default user.
28: invoice emit --from ORD-1
Invoice created: INV-1
Internal (proxy) invoice created: INV-2
Generating payment for INV-2...
Remittance advice (using proxy bank account) created: REM-1
Remittance advice (using business unit bank account) created: REM-2
Invoice sent to client: INV-1
30: as system
Modifying default user.
31: reminder send INV-1
Reminder for invoice sent: INV-1
32: payment match INV-1
Generating payment for INV-1...
Remittance advice (using client bank account) created: REM-3
Remittance advice (using business unit bank account) created: REM-4
